### PR TITLE
Multiple scaffold fixes

### LINKED ISSUE #3218: T4 templates linq2db.SQLite.MS fails to generate views with window functions
In a sqlite db, if a view uses window functions, linq2db.SQLite.MS fails to generate modes. There is an exception `database disk image is malformed` as underneath. But that description is not true. If I remove all views that use window functions, linq2db.SQLite.MS properly generates models .

Here is a simple view to reproduce the issue:
```
CREATE VIEW [MyView]
AS
SELECT 
       [ID], 
       ROW_NUMBER () OVER () AS [Number]
FROM   [MyTable];
```

```
Severity	Code	Description	Project	File	Line	Suppression State
Error		Running transformation: code = Corrupt (11), message = System.Data.SQLite.SQLiteException (0x800007EF): database disk image is malformed
malformed database schema (vMyView) - near "(": syntax error
   at System.Data.SQLite.SQLite3.Prepare(SQLiteConnection cnn, String strSql, SQLiteStatement previous, UInt32 timeoutMS, String& strRemain)
   at System.Data.SQLite.SQLiteCommand.BuildNextCommand()
   at System.Data.SQLite.SQLiteCommand.GetStatement(Int32 index)
   at System.Data.SQLite.SQLiteDataReader.NextResult()
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader()
   at System.Data.SQLite.SQLiteConnection.Schema_Tables(String strCatalog, String strTable, String strType)
   at System.Data.SQLite.SQLiteConnection.GetSchema(String collectionName, String[] restrictionValues)
   at System.Data.SQLite.SQLiteConnection.GetSchema(String collectionName)
   at LinqToDB.DataProvider.SQLite.SQLiteSchemaProvider.GetTables(DataConnection dataConnection, GetSchemaOptions options)
   at LinqToDB.SchemaProvider.SchemaProviderBase.GetSchema(DataConnection dataConnection, GetSchemaOptions options)
   at LinqToDB.DataProvider.SQLite.SQLiteSchemaProvider.GetSchema(DataConnection dataConnection, GetSchemaOptions options)
   at Microsoft.VisualStudio.TextTemplating424180F073C29AFFD48A1375AF38A739BE0DB16D25ADB6B02356F765586ACE97B940FDA0FE99B22C970DE6B3D5BB2B51F474D9D6A432A6A3A3174D3DAF9E177F.GeneratedTextTransformation.LoadServerMetadata(DataConnection dataConnection) in C:\Users\MYNAME\.nuget\packages\linq2db.sqlite.ms\3.4.4\build\..\contentFiles\any\any\LinqToDB.Templates\DataModel.ttinclude:line 111
   at Microsoft.VisualStudio.TextTemplating424180F073C29AFFD48A1375AF38A739BE0DB16D25ADB6B02356F765586ACE97B940FDA0FE99B22C970DE6B3D5BB2B51F474D9D6A432A6A3A3174D3DAF9E177F.GeneratedTextTransformation.LoadMetadata(DataConnection dataConnection) in C:\Users\MYNAME\.nuget\packages\linq2db.sqlite.ms\3.4.4\build\..\contentFiles\any\any\LinqToDB.Templates\DataModel.ttinclude:line 433
   at Microsoft.VisualStudio.TextTemplating424180F073C29AFFD48A1375AF38A739BE0DB16D25ADB6B02356F765586ACE97B940FDA0FE99B22C970DE6B3D5BB2B51F474D9D6A432A6A3A3174D3DAF9E177F.GeneratedTextTransformation.LoadSQLiteMetadata(String connectionString) in C:\Users\MYNAME\.nuget\packages\linq2db.sqlite.ms\3.4.4\build\..\contentFiles\any\any\LinqToDB.Templates\LinqToDB.SQLite.ttinclude:line 37
   at Microsoft.VisualStudio.TextTemplating424180F073C29AFFD48A1375AF38A739BE0DB16D25ADB6B02356F765586ACE97B940FDA0FE99B22C970DE6B3D5BB2B51F474D9D6A432A6A3A3174D3DAF9E177F.GeneratedTextTransformation.TransformText() in D:\project\GIAPHA\1.0\SOURCE\GBE\GBE.Server\GBE.Data\LinqToDB.Models\GBE.tt:line 21		C:\Users\MYNAME\.nuget\packages\linq2db.sqlite.ms\3.4.4\build\..\contentFiles\any\any\LinqToDB.Templates\DataModel.ttinclude	111	
```

### LINKED ISSUE #3762: Scaffold generator generating incorrect attributes for composite FK associations and MSSQL schema-bound functions
### Describe your issue

A large number of tests started failing when we attempted to migrate our project from the old T4 template system to the new scaffold generator.  Investigation of the cause revealed two errors in the attributes being generated by the scaffold generator:

1. Associations generated from <var>n</var>-component composite foreign key constraints join on the first component <var>n</var> times (redundantly) and the other components never—furthermore, all components of `OtherKey` *other than the first* get set to the same name (same `nameof` expression, even) as the corresponding components of `ThisKey`, even when the names do not correspond between the tables.  *In other words*, all components of both `ThisKey` and `OtherKey` apart from the first seem to be clobbered by duplicates of the first component of `ThisKey`.
2. We have a schema-bound scalar function which we use from LINQ expressions.  SQL Server appears to require it to be referenced with an explicit schema prefix (`dbo.`), which the scaffold was not including (unlike the old T4 system).  Furthermore, the scaffold attribute was wrapping the name in `[]`, resulting in `[…](…)` syntax in the generated SQL, which was causing a syntax error on the SQL Server side.

### Steps to reproduce

https://github.com/megaslice/LDBMinim

### Environment details

`Linq To DB` version: 4.2.0

Database (with version): SQL Server 2019

ADO.NET Provider (with version): Microsoft.Data.SqlClient 4.1.0

Operating system: Windows 10

.NET Version: 6.0

### LINKED ISSUE #3777: [Scaffold] random property sorting without changing the structure of the database
Hello Team
After using `linq2db-cli` for some time, I encountered the problem that the order of properties in the file (without changing the structure of the database) changes after generation. This applies to association properties (table files) and table properties in the DataConnection file. This is very inconvenient when using VCS.

### Steps to reproduce
**database.json**
```
{
  "general": {
    "provider": "SQLServer",
    "connection": "Server=.\\;Database=test;Trusted_Connection=True;Enlist=False;Encrypt=no;",
    "output": "c:\\Projects\\test\\Data\\Models\\",
    "overwrite": true
  },
  "code": {
    "namespace": "test.Data"
  },
  "dataModel": {
    "entity-class-name": {
      "pluralization": "none"
    },
    "entity-context-property-name": {
      "pluralization": "none"
    }
  }
}
```
**cmd**
`dotnet linq2db scaffold -i "C:\Projects\test\database.json"`
### Environment details

`Linq To DB` version: *4.2.0.0*
`linq2db-cli` version: *4.2.0.0*
Database (with version): *SQL Server 2019*
.NET Version: *6*

### LINKED ISSUE #3810: Incorrect Association Creation with Composite Primary Keys
### Describe your issue

Hello. There is a problem related to composite primary keys. When we are using scaffolding via the CLI, attribute generation for associations is incorrect. Duplicated key values appears in the attribute. the LoadWith<> method does not work because of that .

### Steps to reproduce

1. Create DB with 2 tables:
    - First table with name PKTable and colums Id, AnotherId
    - Second table with name FKTable and colums PKTableId, PKTableAnotherId

2. Create Composite PK for PKTable with columns Id and AnotherId
3. Create Composite FK for FKTable with columns PKTableId and PKTableAnotherId to PKTable to Id and AnotherId
4. Run CLI command dotnet linq2db scaffold -i database.json
5. Fill DB some random data
6. Write any query with LoadWith<> Method
7. Get null on association property

```c#
        [Table("PKTable")]
	public class PkTable
	{
		[Column("Id"       , IsPrimaryKey = true, PrimaryKeyOrder = 0, IsIdentity = true, SkipOnInsert = true, SkipOnUpdate = true)] public int Id        { get; set; } // integer
		[Column("AnotherId", IsPrimaryKey = true, PrimaryKeyOrder = 1                                                             )] public int AnotherId { get; set; } // integer

		#region Associations
		/// <summary>
		/// FKTable_PKTableAnotherId_PKTableId _fkey backreference
		/// </summary>
		[Association(ThisKey = nameof(Id) + "," + nameof(Id), OtherKey = nameof(FkTable.PkTableId) + "," + nameof(Id))]
		public List<FkTable> FkTablePkTableAnotherIdPkTableIdFkeys { get; set; } = null!;
		#endregion
	}
```

```c#

        [Table("FKTable")]
	public class FkTable
	{
		[Column("Id"              , IsPrimaryKey = true, IsIdentity = true, SkipOnInsert = true, SkipOnUpdate = true)] public int  Id               { get; set; } // integer
		[Column("PKTableId "                                                                                        )] public int? PkTableId        { get; set; } // integer
		[Column("PKTableAnotherId"                                                                                  )] public int? PkTableAnotherId { get; set; } // integer

		#region Associations
		/// <summary>
		/// FKTable_PKTableAnotherId_PKTableId _fkey
		/// </summary>
		[Association(ThisKey = nameof(PkTableId) + "," + nameof(PkTableId), OtherKey = nameof(PkTable.Id) + "," + nameof(PkTableId))]
		public PkTable? PkTableAnotherIdPkTableIdFkey { get; set; }
		#endregion
	}

```

### Environment details

`Linq To DB` version: 4.3.0

Database (with version): PostgreSQL 14

[ADO.NET](http://ado.net/) Provider (with version): NpgSql 7.0.0-rc.1

Operating system: Windows 10

.NET Version: 7.0

### Solution suggestion

image
image

### LINKED ISSUE #3820: Scaffolding Issue with .NET6.0 and Customized Interceptor
### Describe your issue

Dear LINQ TO DB Dev Team,

I have issue when using linq2db CLI scaffolding with ***.NET6.0*** with customized interceptor entity.  

I created a new class which is a derivative of *ScaffoldInterceptors* and inserted my source code (see section below *'Interceptor Class'*). In ***cli.json*** file, in the property **[general]**, I added the key **[customize]** with value of built **DLL**. 

I use **net6.0, linq2db 4.3.0 and linq2db.Tools 4.3.0** and when scaffolding, it shows an error. In your documentation you wrote that **customize** option, can be used on *.NET 3.1* or later versions and that it should be compatible with current runtime. After I changed project's target framework on netcoreapp3.1 and JSON **customize** option's value to new built in DLL *".\\bin\\Debug\\netcoreapp3.1\\PROJ.dll"*, everything works smooth and without errors.

### Summary:
When I use the framework .NET6.0 and scaffolding with customized interceptor entity. It fails.
When I use the framework .NET3.1 and scaffolding with customized interceptor entity. Everything runs perfectly fine.

Am I doing something wrong? What is the problem with customized scaffolding in the .NET6.0 environment?

### Steps to reproduce
```
dotnet linq2db scaffold --import "cli.json"
```

### cli.json
```
"general": {
  "output": ".\\Models",
  "overwrite": true,
  "provider": "SQLServer",
  "connection": "<< YOUR DB CONN STRING HERE >>",
  "template": "default",
  "customize": ".\\bin\\Debug\\net6.0\\PROJ.dll"
}
```

### Interceptor Class
```cs
public class Intreceptor : ScaffoldInterceptors
{
    // several use-cases cases of entity customization
    public override void PreprocessEntity(ITypeParser typeParser, EntityModel entityModel)
    {
        // for table with name "alltypes" we cannot recognize separate words to properly generate class name
        // let's modify generated entity class name in model
        if (entityModel.Class.Name.StartsWith("AspNet"))
            entityModel.Class.Name = entityModel.Class.Name.Replace("AspNet", String.Empty);

        // mark column as non-editable
        var creatorColumn = entityModel.Columns.Where(c => c.Metadata.Name == "created_by").FirstOrDefault();
        if (creatorColumn != null)
            creatorColumn.Metadata.SkipOnUpdate = true;
    }
  }
```

### Error
```
Unhandled exception: Unable to load one or more of the requested types.  
Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
...
at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeModule.GetTypes()
   at System.Reflection.Assembly.GetTypes()
   at LinqToDB.CommandLine.ScaffoldCommand.LoadInterceptorsFromAssembly(String sourcePath, Assembly assembly, ScaffoldOptions options)
   at LinqToDB.CommandLine.ScaffoldCommand.LoadInterceptorsFromAssembly(String assemblyPath, ScaffoldOptions options)
   at LinqToDB.CommandLine.ScaffoldCommand.LoadInterceptors(String interceptorsPath, ScaffoldOptions options)
   at LinqToDB.CommandLine.ScaffoldCommand.Execute(CliController controller, String[] rawArgs, Dictionary`2 options, IReadOnlyCollection`1 unknownArgs)
   at LinqToDB.CommandLine.CliController.Execute(String[] args)
   at LinqToDB.Tools.Program.Main(String[] args)
```

### Environment details

Packages:
`Linq To DB` version: *4.3.0*
`Linq To DB Tools` version: *4.3.0*

Database (with version): *SQL Server 2019*

.NET Version: *6.0*

### PR DESCRIPTION
**Pull request by MaceWindu**
- fix #3777 : order scaffolded objects and columns by name
- fix #3810, fix #3762 : fix association generation for composite keys
- fix #3762 (part 2): always generate schema for SQL Server scalar functions
- fix #3820 : add `net6.0` and `net7.0` support to `linq2db.cli`
- fix #3218 : don't try to load already loaded assembly

### CODE REVIEW

**Comment by sdanyliv** [comment]
> order scaffolded objects and columns by name

I think it is better to preserve database order.

**Review by sdanyliv — APPROVED** [review]

File: Source/LinqToDB.Tools/Scaffold/DataModel/DataModelLoader.Entities.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ partial class DataModelLoader
 			ISet<string>     defaultSchemas,
 			IType?           baseType)
 		{
-			var (tableName, isNonDefaultSchema) = ProcessObjectName(table.Name, defaultSchemas);
+			var (tableName, isNonDefaultSchema) = ProcessObjectName(table.Name, defaultSchemas, false);
```
Comment (viceroypenguin) [inline Source/LinqToDB.Tools/Scaffold/DataModel/DataModelLoader.Entities.cs:29]: add parameter name to `false` for clarity

**Review by viceroypenguin — APPROVED** [review]

**Pull request merged**